on hpux, use gcc to link if used to compile

--- configure.in.orig	2007-10-25 10:57:21.000000000 +0200
+++ configure.in	2007-10-25 10:58:02.000000000 +0200
@@ -658,6 +658,7 @@
 	  esac
 	  BLDLIBRARY='-Wl,+b,$(LIBDIR) -L. -lpython$(VERSION)'
 	  RUNSHARED=SHLIB_PATH=`pwd`:${SHLIB_PATH}
+	  INSTSONAME="$LDLIBRARY".$SOVERSION
 	  ;;
     AIX*)
 	  BLDLIBRARY='-Wl,-blibpath:$(LIBDIR) $(LDLIBRARY)'
@@ -1447,7 +1448,11 @@
 		then LDSHARED='$(CC) -shared'
 		else LDSHARED='$(CC) -G';
 		fi ;;
-	hp*|HP*) LDSHARED="ld -b";;
+	hp*|HP*)
+		if test "$GCC" = "yes"
+		then LDSHARED='$(CC) -shared'
+		else LDSHARED="ld -b"
+		fi ;;
 	OSF*) LDSHARED="ld -shared -expect_unresolved \"*\"";;
 	Darwin/1.3*)
 		LDSHARED='$(CC) $(LDFLAGS) -bundle'
